agency/detail/optional.hpp

```diff
@@ -70,7 +70,9 @@ inline void throw_bad_optional_access(const char* what_arg)
 }
 
 
+#if !(defined(__clang__) && defined(__CUDA__))
```

Comment:
I don't know, I just added `!(defined(__clang__) && defined(__CUDA__))` to whatever was there before. I can also add `defined(__CUDACC__)` if needed.  Perhaps a better option is `__NVCC__` rather than `__CUDACC__` as this is nvcc-specific pragma? Please advice.